daemon-tests/tests/main/collaborative_settlement.rs

```diff
@@ -87,6 +91,54 @@ async fn maker_accepts_collab_settlement_after_commit_finality() {
     wait_next_state!(order_id, maker, taker, CfdState::OpenCommitted);
 }
 
+#[otel_test]
+async fn malicious_taker_tries_to_settle_unrelated_order() {
```

Comment:
It's not that bad, but it's true that it's not the nicest test.

I think it might be a good idea to write a test for the `collab_settlement` actors instead, but I haven't given it much thought.